### PR TITLE
Healthcheck showing s3

### PR DESCRIPTION
**Pull request by tatyree**
I am PRing this on its own as it stands alone and establishes a pattern.  It also contains a couple of small changes that resulted from lessons learned with the downloader (dotenv spec usage, and mutation killing outside of/in addition to feature specs).

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -6,6 +6,23 @@
...
+        dependencies: {
+          external: {
+            s3: {
+              S3::REGION.tr('-','_') => S3.status
+            }
+          }
+        }
+      }.to_json
+    end
+
```
Comment (digitalronin) [inline app.rb]: Can we make /status depend on the same stuff as /healthcheck?

Feel free to make that a TODO comment, for now.

Comment (tatyree) [inline app.rb]: @digitalronin I was thinking they should probably just be amalgamated into a single call, but wanted to talk to Leo about it, first.

Comment (digitalronin) [inline app.rb]: That makes sense.

**Comment by tatyree** [comment]
The S3 check, which is probably the most straightforward of all of them, was longer and more complicated than I expected.  As a result, I thought I'd better dial down the ambition for the individual PRs.

**Pull request merged**